### ISSUE TITLE
FT Asynchronous should run on platform thread by default

### ISSUE THREAD

**Issue by danielkec** [helidon-io/helidon#8591]
## Environment Details
* Helidon Version: 4.0.6
* Helidon SE or Helidon MP: Both

----------

## Problem Description
Fault tolerance asynchronous should always run on physical thread by default.
Currently it is not possible to configure MP for platform thread usage.

Cases for offloading with `@Asynchronous`:
* **Blocking code** - offloading VT->VT is not needed
* **Obstructing/long-running code** - offloading VT->VT is not practical as carrier thread from the same pool is used
* **Pinning code** - offloading VT->VT doesn't solve the issue

More complicated cases for VT's may exist, configurable option should exist(it doesn't in 4.0.6 MP).

## Steps to reproduce
```java
    @GET
    @Path("/platform")
    @Asynchronous
    public CompletionStage<ThreadInfo> getOnPlatformThread() {
        var name = Thread.currentThread().getName();
        var virtual = Thread.currentThread().isVirtual();
        ThreadInfo threadInfo = new ThreadInfo(name, virtual);
        return CompletableFuture.completedFuture(threadInfo);
    }
```

```java
➜  offloading-mp curl -s localhost:8080/thread-name/platform | jq
{
  "name": "helidon-ft-1: async",
  "virtual": true
}
```

**Comment by spericas** [comment]
I agree there should be an option for this, but I'm not sure a platform thread should be default; I see people using asynchronous in 4.x without a good understanding of what they are doing, and suddenly creating a bunch of new platform threads may not be ideal. I will take a look.

**Comment by spericas** [comment]
@danielkec One possibility would be to introduce a new annotation to control the executor service to use. Something along these lines:

```java
    @Asynchronous
    @InvokedUsing(SingleThreadExecutorSupplier.class)
    CompletableFuture<String> asyncWithExecutor() {
        called.set(!Thread.currentThread().isVirtual());
        FaultToleranceTest.printStatus("AsynchronousBean::asyncWithExecutor", "success");
        return CompletableFuture.completedFuture("success");
    }

    static class SingleThreadExecutorSupplier implements Supplier<ExecutorService> {

        @Override
        public ExecutorService get() {
            return Executors.newSingleThreadExecutor();
        }
    }
```

**Comment by danielkec** [comment]
@spericas  I like the PR! There is one other thing we should consider. MP FT is designed for runtimes which are sensitive to blocking - all async methods needs to return `CompletionStage`/`@Future`. This is no longer needed in Helidon as blocking is no problem any more, I think we should support `@Asynchronous` also on methods which don't return future.

I thing we should also contest the spec:
```
The method annotated with @Asynchronous must return a Future or a CompletionStage from the java.util.concurrent package. Otherwise, a FaultToleranceDefinitionException occurs.
```

**Comment by spericas** [comment]
@danielkec Maybe we can wait until the spec catches up, otherwise we'll very likely fail some TCKs. Besides that, I'm not sure I agree with you, if you don't mind blocking the calling thread at the time you call the method, then do you really want to use async instead of the same thread? Returning a future gives you the ability to decide when to (potentially) block the calling thread by waiting on the that future at a specific time.

**Comment by spericas** [comment]
After discussing the use case further with @danielkec, we agreed to proceed with a two-pronged approach: (i) an extension for the FT module to support using a different executor service (this issue!) and (ii) a new, simpler annotation in another module to only handle running arbitrary methods in platform threads. There will be a new issue created for (ii).

**Comment by spericas** [comment]
Replaced by #8638
